presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -182,6 +185,9 @@ private void waitForMinimumWorkers()
 
     private void startExecution(QueryExecution queryExecution)
     {
+        if (queryExecution != null && this.getResourceGroupQueryLimits().isPresent()) {
```

Comment:
Why would `queryExecution` be null?

Reply:
It won't be in a real environment- took some discussion with Mayank - it was null in some test cases - this was my initial solution, but now see changes to `TestLocalDispatchQuery` so that is nonnul